api/base/filters.py

```diff
@@ -540,13 +540,27 @@ def param_queryset(self, query_params, default_queryset):
 
         if filters:
             for key, field_names in filters.iteritems():
-                for field_name, data in field_names.iteritems():
-                    operations = data if isinstance(data, list) else [data]
-                    for operation in operations:
-                        if isinstance(queryset, list):
-                            queryset = self.get_filtered_queryset(field_name, operation, queryset)
-                        else:
-                            queryset = self.filter_by_field(queryset, field_name, operation)
+                match = self.QUERY_PATTERN.match(key)
+                fields = match.groupdict()['fields']
+                if len(re.findall(self.FILTER_FIELDS, fields)) > 1:  # This indicates an OR statement
+                    sub_query = Node.objects.none()
+                    for field_name, data in field_names.iteritems():
```

Comment:
This section looks very similar to L557-L563. Can this be dried and possibly independently tested?